src/ml/neural_net/tf_compute_context.cpp

```diff
@@ -125,7 +125,9 @@ tf_compute_context::tf_compute_context() = default;
 tf_compute_context::~tf_compute_context() = default;
 
 size_t tf_compute_context::memory_budget() const {
-  return 0;
+  // TODO: Returns 4GB as that makes sure default batch size is used. 
+  // Do something that makes more sense like MPS later.
+  return 4294967296;
```

Comment:
Have you tested this? Just to be safe, try `4294967296lu`. The `lu` suffix tells the compiler that the integer literal has type `unsigned long`, which should be big enough to contain `2**32`.

Otherwise, C/C++ defaults to `int`, which is not actually big enough to contain `2**32`

Reply:
I will test it on Linux GPU/CPU by making an OD model with Tensorflow.